src/containers/NahmiiWithdraw/index.js

```diff
@@ -0,0 +1,823 @@
+    if (!amountToWithdrawInputRegex.test(value)) return;
+
+    // don't update if is an infeasible amount of Ether (> 100x entire circulating supply as of Aug 2018)
+    if (!isNaN(value) && Number(value) > 10000000000) return;
+
+    if (!isNaN(value)) {
+      this.setState({ amountToWithdraw: new BigNumber(value) });
+    }
+
+    // update the input (this could be an invalid number, such as '12.')
+    this.setState({ amountToWithdrawInput: value });
+  }
+
+  handleGasPriceChange(e) {
+    const { value } = e.target;
+    // allow an empty input to represent 0
+    if (value === '') {
+      this.setState({ gasPriceGwei: new BigNumber('0'), gasPriceGweiInput: '' });
+    }
+
+    // don't update if invalid regex
+    // (numbers followed by at most 1 . followed by at most 9 decimals)
+    if (!gweiRegex.test(value)) return;
+
+    // don't update if a single gas is an infeasible amount of Ether
+    // (> 100x entire circulating supply as of Aug 2018)
+    if (!isNaN(value) && Number(value) > 10000000000000000000) return;
+
+    // update the input (this could be an invalid number, such as '12.')
+    this.setState({ gasPriceGweiInput: value });
+
+    // update actual gwei if it's a real number
+    if (!isNaN(value)) {
+      this.setState({ gasPriceGwei: new BigNumber(value) });
+    }
+  }
+
+  handleGasLimitChange(e) {
+    const { value } = e.target;
+    // allow an empty input to represent 0
+    if (value === '') {
+      this.setState({ gasLimitInput: '', gasLimit: 0 });
+    }
+
+    // only allow whole numbers
+    if (!gasLimitRegex.test(value)) return;
+
+    // don't allow infeasible amount of gas
+    // (gas limit per block almost never exeeds 10 million as of Aug 2018  )
+    const ONE_HUNDRED_MILLION = 100000000;
+    if (value > ONE_HUNDRED_MILLION) return;
+
+    this.setState({ gasLimitInput: value, gasLimit: parseInt(value, 10) });
+  }
+
+  generateTransferingStatus() {
+    const {
+      currentWalletWithInfo,
+      withdrawals,
+      ongoingChallenges,
+      settleableChallenges,
+      currentNetwork,
+      intl,
+      ledgerNanoSInfo,
+      trezorInfo,
+    } = this.props;
+    const { formatMessage } = intl;
+    const confOnDevice = ledgerNanoSInfo.get('confTxOnDevice') || trezorInfo.get('confTxOnDevice');
+    let ratio;
+    let type;
+    const activities = [withdrawals, ongoingChallenges, settleableChallenges];
+    for (const activity of activities) { //eslint-disable-line
+      if (activity.get('status') === 'requesting') ratio = '1/2';
+      if (activity.get('status') === 'mining') ratio = '2/2';
+      if (activity.get('status') === 'receipt') ratio = '2/2';
+
+      if (ratio && activity === withdrawals) type = 'withdraw';
+      if (ratio && activity === ongoingChallenges) type = 'start_challenge';
+      if (ratio && activity === settleableChallenges) type = 'confirm_settle';
+
+      if (ratio) break;
+    }
+
+    if (!ratio) return null;
+    const transferingText =
```

Comment:
It looks like when the ratio is 1/2 it's not waiting for a transaction to be mined, but some other type of operation. Maybe the text can be updated here to reflect that

Reply:
Yes, that is better.